File: backend/src/schemas/reminder.py
"""
Reminder Pydantic Schemas
Feature: 005-advanced-features-dapr-kafka
"""

from pydantic import BaseModel, field_validator
from typing import Optional
from uuid import UUID
from datetime import datetime


class ReminderCreateRequest(BaseModel):
    offset_minutes: int

    @field_validator('offset_minutes')
    @classmethod
    def validate_offset(cls, v: int) -> int:
        if v <= 0:
            raise ValueError('offset_minutes must be greater than 0')
        return v


class ReminderResponse(BaseModel):
    id: UUID
    offset_minutes: int
    trigger_at: datetime
    status: str

    class Config:
        from_attributes = True

File: frontend/types/reminder.ts
/**
 * Reminder Type Definitions
 * Feature: 005-advanced-features-dapr-kafka
 */

export interface Reminder {
  id: string;
  offset_minutes: number;
  trigger_at: string;
  status: 'pending' | 'sent' | 'cancelled';
}

export interface ReminderCreatePayload {
  offset_minutes: number;
}
